### PR TITLE
PLANET-3266 search

### PR DESCRIPTION
**Pull request by angtheod**
Switch to ElasticPress and remove old P4_Search that supported both SearchWP and EP.

Now that P4 connects to our elasticsearch cluster (see [PR](https://github.com/greenpeace/planet4-base-fork/pull/20)), we can use the P4_ElasticSearch class.

Related PRs:
#391 #401 #411 #436 #706 #710 

[Test env](https://k8s.p4.greenpeace.org/search/?s=&orderby=_score)

### CODE REVIEW

File: templates/navigation-bar.twig [hunk 1]
```diff
@@ -60,9 +60,12 @@
 			<span class="screen-reader-text">{{ __( 'Toggle search form', 'planet4-master-theme' ) }}</span>
 		</button>
 		<form id="search_form" action="{{ data_nav_bar.home_url }}" class="form nav-item nav-search-wrap">
-			<input type="text" class="form-control" placeholder="{{ search_label }}"
+			<input id="search_input" type="text" class="form-control" placeholder="{{ search_label }}"
 				   value="{{ data_nav_bar.search_query|e('wp_kses_post')|raw }}" name="s" aria-label="Search">
 			<input id="orderby" type="hidden" name="orderby" value="{{ selected_sort ?? default_sort }}"/>
+			{% if true == is_elastic_search %}
```
Comment (kirdia) [inline templates/navigation-bar.twig]: This is probably a leftover from a previous commit, I don't see it used anywhere.

File: classes/class-p4-elasticsearch.php [hunk 2]
```diff
@@ -336,6 +328,8 @@ protected function get_posts( $paged = 1 ) : array {
 								];
 								break;
 							case 'ptype':
+								// This taxonomy is used only for Posts.
+								$args['post_type']   = 'post';
```
Comment (kirdia) [inline classes/class-p4-elasticsearch.php]: The whole 'filtering check' could be extracted to it's own function, to unclutter the function. What do you think @angtheod ?
https://github.com/greenpeace/planet4-master-theme/blob/f65a0dd355a66609cf1eafc637bee2167b2d679f/classes/class-p4-elasticsearch.php#L312

File: classes/class-p4-elasticsearch.php [hunk 3]
```diff
@@ -693,7 +797,7 @@ protected function set_results_context( &$context ) {
 		public function validate( &$selected_sort, &$filters, $context ) : bool {
 			$selected_sort = filter_var( $selected_sort, FILTER_SANITIZE_STRING );
 			if ( ! isset( $context['sort_options'] ) || ! in_array( $selected_sort, array_keys( $context['sort_options'] ), true ) ) {
-				$selected_sort = P4_Search::DEFAULT_SORT;
+				$selected_sort = P4_ElasticSearch::DEFAULT_SORT;
```
Comment (kirdia) [inline classes/class-p4-elasticsearch.php]: Please change to `self` in this case.

**Comment by kirdia** [comment]
Overall looks pretty good!! +1 
I left some comments.
Tested locally and functionality works well. I didn't notice any errors or warnings after indexing elasticsearch.
Can you check the search acceptance test? it fails in the ci. Probably it is because elasticsearch plugin is not setup correctly, but just to be sure it does not break anything.

**Comment by comzeradd** [comment]
Can you also check the failing tests. I can [see](https://circleci.com/gh/greenpeace/planet4-master-theme/1403) 3 `phpcs` errors and 1 `phpunit`.

**Pull request merged**